### ISSUE TITLE
Add TypeScript typings

### ISSUE THREAD

**Issue by msiebuhr** [alexjeffburke/jest-unexpected#11]
I'm trying to use this with typescript, and the lack of types is making the Jest whine quite a bit:
```
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    lib/.../foo.test.ts:1:25 - error TS7016: Could not find a declaration file for module 'jest-unexpected'. '/Users/.../node_modules/jest-unexpected/lib/jestUnexpected.js' implicitly has an 'any' type.
      Try `npm install @types/jest-unexpected` if it exists or add a new declaration (.d.ts) file containing `declare module 'jest-unexpected';`

    1 import expect = require('jest-unexpected');
```

Related to https://github.com/unexpectedjs/unexpected/issues/525

**Comment by msiebuhr** [comment]
Yeah, we're slowly moving to TS. We're writing our new projects in TS, while we're also moving several of our current projects over.

As I now understand TS typings (the files that adds types), it should be sufficient to re-use one from "upstream" unexpected and run with it.

**Comment by alexjeffburke** [comment]
I did some digging and discovered that the jest expect typings are actually defined within the jest definition as a nested `interface`. I can probably inline that but it might affect keeping them in sync. Let me see what I can do.

**Comment by msiebuhr** [comment]
It is my impression that TS spots the `jest-unexpected` of `const expect = require(jest-unexpected)` and then go looking for `jest-unexpected.d.ts` locally, a `typings = <filename.d.ts>` in jest-unexpcected's package.json and also an `@types/jest-unexpected` package.

I haven't checked the order, but fulfilling any of those should do it.
